.github/workflows/publish-internal.yml

```diff
@@ -0,0 +1,136 @@
+        type: choice
+        options:
+          - DotNetFoundation
+          - Riganti
+        default: DotNetFoundation
+        description: The signature to be used to sign the packages.
+        required: false
+
+jobs:
+  read-input:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-version
+        run: >
+          if [ "${{ inputs.release-type }}" == 'InternalPreview' ]; then
+            VERSION="${{ inputs.version-core}}-${{ inputs.prerelease-version }}-${{ github.run_id }}";
+          elif [ "${{ inputs.release-type }}" == 'PublicPreview' ]; then
+            VERSION="${{ inputs.version-core}}-${{ inputs.prerelease-version }}-final";
+          elif [ "${{ inputs.release-type }}" == 'Stable' ]; then
+            VERSION="${{ inputs.version-core}}-${{ inputs.prerelease-version }}";
+          else
+            echo "Unknown release type '${{ inputs.release-type }}'.";
+            exit 1;
+          fi;
+          echo "version=$VERSION" >> $GITHUB_OUTPUT;
+    outputs:
+      version: ${{ steps.set-version.outputs.version }}
+
+  publish-nuget-packages:
+    runs-on: windows-2022
+    needs: read-input
+    steps:
+
+    - uses: actions/checkout@v3
+
+    - name: Set up
+      uses: ./.github/setup
+
+    - name: Prepare signclient config
+      if: ${{ inputs.signature-type == 'DotNetFoundation' }}
+      run: Write-Output "$env:SIGN_CONFIG" | Out-File "${{ github.workspace }}/signconfig.json"
+      env:
+        SIGN_CONFIG: ${{ secrets.SIGN_CONFIG }}
+      working-directory: ${{ github.workspace }}/src
+
+    - name: Add internal NuGet feed
+      run: ./ci/scripts/Add-InternalNuGetFeed.ps1 `
+          -internalFeed "${{ secrets.AZURE_ARTIFACTS_FEED }}" `
+          -internalFeedUser "${{ secrets.AZURE_ARTIFACTS_USERNAME }}" `
+          -internalFeedPat "${{ secrets.AZURE_ARTIFACTS_PAT }}"
+
+    - name: Publish NuGet packages (.NET Foundation)
+      if: ${{ inputs.signature-type == 'DotNetFoundation' }}
+      run: ./ci/scripts/Publish-NuGetPackages.ps1 `
+          -root "${{ github.workspace }}" `
+          -version "${{ needs.read-input.outputs.version }}" `
+          -signatureType "DotNetFoundation" `
+          -dnfUser "${{ secrets.SIGN_USER }}" `
+          -dnfSecret "${{ secrets.SIGN_SECRET }}"
+
+    - name: Publish NuGet packages (Riganti)
+      if: ${{ inputs.signature-type == 'Riganti' }}
+      run: ./ci/scripts/Publish-NuGetPackages.ps1 `
+          -root "${{ github.workspace }}" `
+          -version "${{ needs.read-input.outputs.version }}" `
+          -signatureType "Riganti" `
+          -rigantiUrl "${{ secrets.SIGN_RIGANTI_KEYVAULT_URL }}" `
+          -rigantiClientId "${{ secrets.SIGN_RIGANTI_CLIENT_ID }}" `
+          -rigantiTenantId "${{ secrets.SIGN_RIGANTI_TENANT_ID }}" `
+          -rigantiSecret "${{ secrets.SIGN_RIGANTI_SECRET }}" `
+          -rigantiCertificate "${{ secrets.SIGN_RIGANTI_CERTIFICATE_NAME }}"
+
+  publish-dotvvm-types:
```

Comment:
We also need to add `dotvvm-jscomponent-react` package, but we can do it in separate PR